tests/integration/test_storage_kerberized_kafka/test.py

```diff
@@ -0,0 +1,143 @@
+from kafka.protocol.admin import DescribeGroupsResponse_v1, DescribeGroupsRequest_v1
+from kafka.protocol.group import MemberAssignment
+import socket
+
+cluster = ClickHouseCluster(__file__)
+instance = cluster.add_instance('instance',
+                                main_configs=['configs/kafka.xml', 'configs/log_conf.xml' ],
+                                with_kerberized_kafka=True,
+                                )
+kafka_id = ''    # instance.cluster.kafka_docker_id
+
+# Helpers
+
+def check_kafka_is_available():
+
+    # plaintext
+    p = subprocess.Popen(('docker',
+                          'exec',
+                          '-i',
+                          kafka_id,
+                          '/usr/bin/kafka-broker-api-versions',
+                          '--bootstrap-server',
+                          'localhost:9093'),
+                         stdout=subprocess.PIPE)
+    p.communicate()
+    return p.returncode == 0
+
+
+def wait_kafka_is_available(max_retries=50):
+    retries = 0
+    while True:
+        if check_kafka_is_available():
+            break
+        else:
+            retries += 1
+            if retries > max_retries:
+                raise "Kafka is not available"
+            print("Waiting for Kafka to start up")
+            time.sleep(1)
+
+
+def kafka_produce(topic, messages, timestamp=None):
+    producer = KafkaProducer(bootstrap_servers="localhost:9093")
+    for message in messages:
+        producer.send(topic=topic, value=message, timestamp_ms=timestamp)
+        producer.flush()
+    print ("Produced {} messages for topic {}".format(len(messages), topic))
+
+
+
+# Fixtures
+
+@pytest.fixture(scope="module")
+def kafka_cluster():
+    try:
+        global kafka_id
+        cluster.start()
+        kafka_id = instance.cluster.kerberized_kafka_docker_id
+        print("kafka_id is {}".format(kafka_id))
+        yield cluster
+
+    finally:
+        cluster.shutdown()
+
+
+@pytest.fixture(autouse=True)
+def kafka_setup_teardown():
+    instance.query('DROP DATABASE IF EXISTS test; CREATE DATABASE test;')
+    wait_kafka_is_available()
+    print("kafka is available - running test")
+    yield  # run test
+
+# Tests
+
+@pytest.mark.timeout(180)  # wait to build containers
+def test_kafka_json_as_string(kafka_cluster):
+    kafka_produce('kafka_json_as_string', ['{"t": 123, "e": {"x": "woof"} }', '', '{"t": 124, "e": {"x": "test"} }', '{"F1":"V1","F2":{"F21":"V21","F22":{},"F23":"V23","F24":"2019-12-24T16:28:04"},"F3":"V3"}'])
+
+    instance.query('''
+        CREATE TABLE test.kafka (field String)
+            ENGINE = Kafka
+            SETTINGS kafka_broker_list = 'kerberized_kafka1:19092',
+                     kafka_topic_list = 'kafka_json_as_string',
+                     kafka_group_name = 'kafka_json_as_string',
+                     kafka_format = 'JSONAsString',
+                     kafka_flush_interval_ms=1000;
+        ''')
+
+    result = instance.query('SELECT * FROM test.kafka;')
```

Comment:
So librdkafka do kinit automatically?

Reply:
will it be ok it num_consumers will be bigger / just several kafka tables will work? (all of them will run kinit?)

Reply:
https://stackoverflow.com/questions/51207555/parallel-kinit-calls-lead-to-a-corrupted-kerberos-cache

Reply:
or librdkafka deals with that somehow / with some locking mechanisms?

Reply:
librdkafka calls kinit, it has locks, while I will recheck that no race conditions here.

Reply:
Locks look pretty simple and reliable.

Reply:
OK